index.html

```diff
@@ -2345,17 +2345,43 @@ <h3>
           are given by the following algorithm. The algorithm takes a
           <a data-cite=
           "WEBIDL#sequence-type">sequence</a>&lt;<a>ShortcutItem</a>&gt;
-          <var>shortcuts</var> as an argument. This algorithm returns an
-          <a data-cite=
-          "WEBIDL#sequence-type">sequence</a>&lt;<a>ShortcutItem</a>&gt;. For
-          each <var>shortcut</var> (<a>ShortcutItem</a>) in the sequence, set
-          <var>shortcut.icons</var> to the result of running <a>processing
-          `ImageResource` members</a> given <var>shortcut.icons</var> and
-          <var>manifest URL</var>. For each <var>shortcut</var>
-          (<a>ShortcutItem</a>) in the sequence, parse
-          <var>shortcut</var>["<a>url</a>"] using <var>manifest URL</var> as
-          the base URL.
+          <var>shortcuts</var> and a <a>URL</a> <var>manifest URL</var>. This
+          algorithm returns a <a data-cite=
+          "WEBIDL#sequence-type">sequence</a>&lt;<a>ShortcutItem</a>&gt;.
         </p>
+        <ol>
+          <li>Let <var>processedShortcuts</var> be a new Array object created
+          as if by the expression [].
+          </li>
+          <li>For each <var>shortcut</var> (<a>ShortcutItem</a>) in the
+          sequence:
+            <ol>
+              <li>If <var>shortcut</var>["name"] or <var>shortcut</var>["url"]
+              are undefined, <a>issue a developer warning</a> and
+              <a>continue</a>.
```

Comment:
Undefined or the empty string perhaps?

Reply:
I thought about this, but for consistency we don't seem to check for empty string anywhere else in the file (e.g., for the `name` field at the top level). Do you think it's worth inconsistent treatment here?

Reply:
We do check for empty strings where it makes sense, like the [Web Share Target spec](https://wicg.github.io/web-share-target/#share_target-member). I think it makes sense here.

We don't do it elsewhere because manifest properties are interpretable without `name`, so we don't want to invalidate the whole manifest when its `name` is empty. In this case, it's seems to be implied that if the `name` of a `ShortcutItem` is the empty string, it's invalid.

Reply:
Agreed on warnings for missing and empty. We were actually talking about this internally as we build out the implementation in Windows.

Reply:
OK. Added "or if name is the empty string". Web Share Target is a reasonable precedent (even though it's not in the manifest spec yet).